src/github.com/t3chguy/matrix-static/matrix-static.go

```diff
@@ -84,12 +88,16 @@ func main() {
 
 	log.Infof("Matrix-Static (%+v)", config)
 
-	client, err := mxclient.NewClient(config.ConfigFile)
+	client, err := mxclient.NewClient(config.MediaBaseURL, config.ConfigFile)
 	if err != nil {
 		log.WithError(err).Error("Unable to start new Client")
 		return
 	}
 
+	if config.MediaBaseURL == "" {
+		config.MediaBaseURL = client.HomeserverURL.String()
+	}
```

Comment:
Why not moving this a few lines up (somewhere before mxclient.NewClient) and remove [this](https://github.com/t3chguy/matrix-static/pull/59/files#diff-a342b5501fd6aed8e49147d8030ad39dR130) for deduplication?

Reply:
because `client.HomeserverURL` is not defined until after `mxclient.NewClient`, and `NewClient` is what reads the config file which holds that value, unless im missing something?